platform/ios/src/MGLMapView.mm

```diff
@@ -4231,6 +4361,84 @@ - (void)updateUserLocationAnnotationView
     [self updateUserLocationAnnotationViewAnimatedWithDuration:0];
 }
 
+- (void)updateAnnotationViews
+{
+    BOOL delegateImplementsViewForAnnotation = [self.delegate respondsToSelector:@selector(mapView:viewForAnnotation:)];
+    
+    if (!delegateImplementsViewForAnnotation)
+    {
+        return;
+    }
+    
+    // Update all visible annotation views
+    std::set<MGLAnnotationTag> visibleTags;
+    std::vector<MGLAnnotationTag> annotationTags = [self annotationTagsInRect:CGRectInset(self.bounds, -_largestAnnoationViewWidth, -_largestAnnotationViewHeight)];
+    
+    for(auto const& annotationTag: annotationTags)
+    {
+        auto &annotationContext = _annotationContextsByAnnotationTag[annotationTag];
+        id<MGLAnnotation> annotation = annotationContext.annotation;
+        
+        CGPoint annotationPoint = [self convertCoordinate:annotation.coordinate toPointToView:self];
+       
+        // If there is no annotation view at this point, it means the context's view was reused by some
+        // other context so we need to reuse or make a new view.
+        if (!annotationContext.annotationView)
+        {
+            MGLAnnotationView *annotationView = [self annotationViewForAnnotation:annotation];
+            
+            if (annotationView)
+            {
+                // If the annotation view has no superview it means it was never used before so add it
+                if (!annotationView.superview)
+                {
+                    [self.glView addSubview:annotationView];
+                }
+                
+                annotationContext.annotationView = annotationView;
+            }
+        }
+       
+        annotationContext.annotationView.center = annotationPoint;
+        annotationContext.annotationView.hidden = annotationContext.annotationView.hidden ?: NO;
```

Comment:
> Tail work might be to investigate why `annotationTagsInRect` (and the underlying boost query) does not return some tags that are definitely visible when the map is tilted  (which is why we are currently hiding as a workaround)

When the map is tilted, the visible region isn’t really a rectangle: https://github.com/mapbox/mapbox-gl-native/issues/2259#issuecomment-149659328. Depending on the rotation, `visibleCoordinateBounds` may end up being inscribed in or circumscribed around the actual visible region. I think this PR can land without handling that situation perfectly, but we’ll definitely want to revisit before release.